File: backend/app/schemas/leads.py
"""Lead motoru (BSD kuyrugu / otonom davet) ekrani sozlesmeleri.

Tum parasal alanlar TRY'ye normalize edilmistir ve `_try` ile biter.
"""

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field


class LeadQueueItem(BaseModel):
    """Danisman ekranindaki tek bir satir (BSD/otonom/dislanan)."""

    user_id: int
    first_name: str
    last_name: str
    email: str
    decision: str = Field(description="BSD | AUTONOMOUS | EXCLUDED")
    exclusion_reason: str | None = Field(
        default=None, description="Yalnizca decision=EXCLUDED icin dolu"
    )
    score: int = Field(description="0-100 potansiyel skoru, siralama icin")
    score_components: dict = Field(default_factory=dict)
    reasons: list[str] = Field(default_factory=list, description="Turkce gerekceler")
    total_value_try: float
    monthly_income: float
    likit_para: float
    phone_number: str | None = None
    birth_date: date | None = Field(
        default=None, description="Yas EKRANDA bundan turetilir, ayrica saklanmaz"
    )
    tckn_last4: str | None = Field(default=None, description="Arayuzde '•••• 1234' gosterimi")
    registered_at: datetime | None = Field(
        default=None,
        description="Musterinin sisteme eklendigi an (`users.created_at`). "
        "Satirin `created_at` alaniyla KARISTIRILMAMALI: o, kaydin hangi "
        "taramada/temasta uretildigini soyler ve her taramada degisir.",
    )
    days_since_activity: int | None = None
    call_outcome: str | None = Field(
        default=None,
        description="Danismanin son isaretledigi gorusme sonucu: "
        "KABUL | ISTEMIYOR | ULASILAMADI. Hic isaretlenmediyse (ya da "
        "'ACIK' ile temizlendiyse) None.",
    )
    call_outcome_at: datetime | None = Field(
        default=None, description="Yukaridaki sonucun isaretlendigi an"
    )
    mail_gonderildi: bool = Field(
        default=False,
        description="Otonom kuyrukta anlamli: mail fiilen gonderildi mi, yoksa "
        "kota/hata freni/Gmail ayari nedeniyle beklemede mi",
    )
    created_at: datetime


class LeadScanSummary(BaseModel):
    """Bir taramanin ozeti - kuyruk ekranlarinin ust bilgisi."""

    scan_id: int | None = None
    trigger: str | None = Field(default=None, description="startup | manual | test")
    started_at: datetime | None = None
    finished_at: datetime | None = None
    scanned_count: int = 0
    bsd_count: int = 0
    autonomous_count: int = 0
    excluded_count: int = 0
    emailed_count: int = 0
    skipped: bool = Field(default=False, description="Asgari araliktan dolayi atlandi mi")
    skip_reason: str | None = None


class LeadQueueResponse(BaseModel):
    items: list[LeadQueueItem]
    count: int
    scan: LeadScanSummary


class LeadScanRequest(BaseModel):
    force: bool = Field(default=False, description="Asgari aralik kontrolunu atla")


class LeadOutcomeRequest(BaseModel):
    """Danismanin telefon gorusmesinden sonra elle isaretledigi sonuc.

    `ACIK` bir sonuc DEGIL, "sonucu temizle" demektir: yanlis isaretlemeyi
    satir silmeden geri alir (`lead_call_outcomes` ekleme-only'dir).
    """

    outcome: Literal["KABUL", "ISTEMIYOR", "ULASILAMADI", "ACIK"]
    note: str | None = Field(default=None, max_length=500, description="Serbest not")

File: frontend/src/models/leads.ts
export type LeadQueueItem = {
  user_id: number;
  first_name: string;
  last_name: string;
  email: string;
  decision: string;
  exclusion_reason: string | null;
  score: number;
  score_components: Record<string, number>;
  reasons: string[];
  total_value_try: number;
  monthly_income: number;
  likit_para: number;
  phone_number: string | null;
  /** Yas EKRANDA bundan turetilir (bkz. `yasHesapla`), ayrica saklanmaz. */
  birth_date: string | null;
  tckn_last4: string | null;
  /**
   * Musterinin sisteme eklendigi an (`users.created_at`). Asagidaki
   * `created_at` ile KARISTIRILMAMALI: o, satirin hangi taramada/temasta
   * uretildigini soyler ve her taramada degisir.
   */
  registered_at: string | null;
  days_since_activity: number | null;
  /** Danismanin son isaretledigi gorusme sonucu; hic isaretlenmediyse null. */
  call_outcome: CallOutcome | null;
  call_outcome_at: string | null;
  mail_gonderildi: boolean;
  created_at: string;
};

/**
 * `ACIK` bir sonuc DEGIL, "sonucu temizle" komutudur: yalnizca ISTEKTE
 * gonderilir, yanitta hic gorunmez (backend onu null'a cevirir).
 */
export type CallOutcome = "KABUL" | "ISTEMIYOR" | "ULASILAMADI";
export type CallOutcomeInput = CallOutcome | "ACIK";

export type LeadScanSummary = {
  scan_id: number | null;
  trigger: string | null;
  started_at: string | null;
  finished_at: string | null;
  scanned_count: number;
  bsd_count: number;
  autonomous_count: number;
  excluded_count: number;
  emailed_count: number;
  skipped: boolean;
  skip_reason: string | null;
};

export type LeadQueueResponse = {
  items: LeadQueueItem[];
  count: number;
  scan: LeadScanSummary;
};